src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -138,8 +248,99 @@ pub struct Child {
     pub ast_crumbs : ast::Crumbs,
 }
 
+impl Deref for Child {
+    type Target = Node;
+    fn deref(&self) -> &Self::Target {
+        &self.node
+    }
+}
+
+impl DerefMut for Child {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.node
+    }
+}
```

Comment:
I know we follow this pattern everythere, but I found this recently https://rust-lang.github.io/api-guidelines/predictability.html#only-smart-pointers-implement-deref-and-derefmut-c-deref